contracts/src/Zappers/WETHZapper.sol

```diff
@@ -224,19 +205,36 @@ contract WETHZapper is AddRemoveManagers, LeftoversSweep {
         address payable _receiver,
         InitialBalances memory _initialBalances
     ) internal {
-        // WETH -> ETH
-        if (!_isCollIncrease) {
-            WETH.withdraw(_collChange);
-            (bool success,) = _receiver.call{value: _collChange}("");
-            require(success, "WZ: Sending ETH failed");
-        }
         // Send Bold
         if (_isDebtIncrease) {
             boldToken.transfer(_receiver, _boldChange);
         }
 
+        // WETH -> ETH
+        uint256 ethToSend;
+        if (!_isCollIncrease) {
+            ethToSend = _collChange;
+        }
+
         // return leftovers to user
-        _returnLeftovers(WETH, boldToken, _initialBalances);
+        uint256 currentBoldBalance = boldToken.balanceOf(address(this));
+        if (currentBoldBalance > _initialBalances.boldBalance) {
+            boldToken.transfer(_initialBalances.sender, currentBoldBalance - _initialBalances.boldBalance);
+        }
+
+        uint256 currentCollBalance = WETH.balanceOf(address(this));
+        if (currentCollBalance > ethToSend + _initialBalances.collBalance) {
+            ethToSend = ethToSend + currentCollBalance - ethToSend - _initialBalances.collBalance;
```

Comment:
In other words: `currentCollBalance - _initialBalances.collBalance`.
Why the `ethToSend - ethToSend`?

Reply:
Moreover: in the case of BOLD, we treat newly minted BOLD differently from "leftover" BOLD, in that leftovers are sent back to the original `msg.sender` while newly minted BOLD is sent to the receiver. This makes sense.

However, in the case of (W)ETH, all of it is sent to the receiver, even leftovers. Can there be leftovers in the first place? I don't think so, which would make this a non-issue, but in that case the code is more complicated than it needs to be.

Reply:
:man_facepalming:

Reply:
Should be fixed now.